### ISSUE TITLE
ODC produces deprecation warning when SQLAlchemy>=1.4 is installed

### ISSUE THREAD

**Issue by SpacemanPaul** [opendatacube/datacube-core#1065]
Against postgresql 12, odc 1.8.3 produces the following warning on initialisation:

```
$ python 
Python 3.6.9 (default, Oct  8 2020, 12:12:24) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import datacube
>>> dc = datacube.Datacube()
/env/lib/python3.6/site-packages/datacube/drivers/postgres/_connections.py:84: SADeprecationWarning:
Calling URL() directly is deprecated and will be disabled in a future release.  The public constructor for 
URL is now the URL.create() method.
  username=username, password=password,
```

Upgrading psycopg2 and sql-alchemy to latest versions didn't fix it.

**Comment by Kirill888** [comment]
I doubt it's issue with server db version, it's probably just SQLAclhemy change in API
https://github.com/opendatacube/datacube-core/blob/afca22831eecf8092e0c293064a9ea465aa6d778/datacube/drivers/postgres/_connections.py#L23
https://github.com/opendatacube/datacube-core/blob/afca22831eecf8092e0c293064a9ea465aa6d778/datacube/drivers/postgres/_connections.py#L84-L88

This is logged before database connection was established.

**Comment by Kirill888** [comment]
@SpacemanPaul downgrade `SQLAclhemy==1.3.23` or disable that warning manually, suggested alterantive `URL.create` was added in 1.4.*, so we can't have simple and compatible URL construction for different SQLAlchemy versions.

**Comment by Kirill888** [comment]
this has been fixed in #1091
